app/src/main/java/com/oracle/javafx/scenebuilder/app/util/TextInputControlPaster.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022, Gluon and/or its affiliates.
+ * All rights reserved. Use is subject to license terms.
+ *
+ * This file is available and licensed under the following license:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the distribution.
+ *  - Neither the name of Oracle Corporation and Gluon nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.oracle.javafx.scenebuilder.app.util;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import javafx.scene.control.IndexRange;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.input.Clipboard;
+
+public class TextInputControlPaster {
+    private final TextInputControl tic;
+    private final Supplier<String> textSource;
+    
+    /**
+     * This class provides a workaround for the issue in JavaFX
+     * which prevents Scene Builder from pasting text values properly
+     * in TextInputControl elements on MacOS. If attempted, clip board
+     * contents is always pasted twice.
+     * 
+     * @see https://bugs.openjdk.java.net/browse/JDK-8205915
+     */
+    public TextInputControlPaster(TextInputControl tic) {
+        this(tic, ()->Clipboard.getSystemClipboard().toString());
```

Comment:
Testing on Mac (Monterey), if I copy the text "1234" into the clipboard, select a label node, and paste it into the `Text` field in the Inspector tab I get without this PR the expected `12341234`, but with this PR I get: 
```
1234javafx.scene.input.Clipboard@5474435e
```
Changing  `() -> Clipboard.getSystemClipboard().toString()` into `() -> ""` seems to prevent that. I wonder why you have this?

In any case, I get the caret at the beginning of the text, which is inconvenient.

Reply:
Also this code has been removed. It's not needed.